python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -568,6 +570,29 @@ def agg(self, func):
 
         if self._sort:
             result = result.sort_index()
+        else:
+            if cudf.get_option(
+                "mode.pandas_compatible"
+            ) and not libgroupby._is_all_scan_aggregate(normalized_aggs):
+                # Even with `sort=False`, pandas clearly documents and
+                # ensures that the ordering inside the group by columns
+                # is preserved(i.e., it is deterministic).
```

Comment:
Let's just use the same language Pandas does:


```suggestion
                # Even with `sort=False`, pandas guarantees that
                # groupby preserves the order of rows within each group.
```